From f4146208962c8463648952cab11b1cdda87ef34f Mon Sep 17 00:00:00 2001
From: Sushanth Sowmyan <khorgath@apache.org>
Date: Thu, 29 Jan 2015 18:45:28 +0000
Subject: [PATCH 0007/1164] HIVE-9501 : DbNotificationListener doesn't include
 dbname in create database notification and does
 not include tablename in create table
 notification (Alan Gates via Sushanth Sowmyan)

git-svn-id: https://svn.apache.org/repos/asf/hive/trunk@1655793 13f79535-47bb-0310-9956-ffa450edef68
(cherry picked from commit 8a29303b07651db86de15bc1374c1ab0efb7a07c)
---
 .../hcatalog/listener/DbNotificationListener.java  |    7 ++-----
 .../hcatalog/api/TestHCatClientNotification.java   |    4 ++--
 .../listener/TestDbNotificationListener.java       |    4 ++--
 3 files changed, 6 insertions(+), 9 deletions(-)

diff --git a/hcatalog/server-extensions/src/main/java/org/apache/hive/hcatalog/listener/DbNotificationListener.java b/hcatalog/server-extensions/src/main/java/org/apache/hive/hcatalog/listener/DbNotificationListener.java
index 3ea2827..9b14851 100644
--- a/hcatalog/server-extensions/src/main/java/org/apache/hive/hcatalog/listener/DbNotificationListener.java
+++ b/hcatalog/server-extensions/src/main/java/org/apache/hive/hcatalog/listener/DbNotificationListener.java
@@ -118,8 +118,7 @@ public void onCreateTable (CreateTableEvent tableEvent) throws MetaException {
     NotificationEvent event = new NotificationEvent(0, now(),
         HCatConstants.HCAT_CREATE_TABLE_EVENT, msgFactory.buildCreateTableMessage(t).toString());
     event.setDbName(t.getDbName());
-    // Table name is not set in create table because this goes on the queue for the database the
-    // table is created in, not the (new) queue for the table itself.
+    event.setTableName(t.getTableName());
     enqueue(event);
   }
 
@@ -208,9 +207,7 @@ public void onCreateDatabase (CreateDatabaseEvent dbEvent) throws MetaException
     NotificationEvent event = new NotificationEvent(0, now(),
         HCatConstants.HCAT_CREATE_DATABASE_EVENT,
         msgFactory.buildCreateDatabaseMessage(db).toString());
-    // Database name is null for create database, because this doesn't belong to messages for
-    // that database.  Rather it belongs to system wide messages.  The db name is in the message,
-    // so listeners can determine it.
+    event.setDbName(db.getName());
     enqueue(event);
   }
 
diff --git a/itests/hcatalog-unit/src/test/java/org/apache/hive/hcatalog/api/TestHCatClientNotification.java b/itests/hcatalog-unit/src/test/java/org/apache/hive/hcatalog/api/TestHCatClientNotification.java
index d730eca..7d7e7b1 100644
--- a/itests/hcatalog-unit/src/test/java/org/apache/hive/hcatalog/api/TestHCatClientNotification.java
+++ b/itests/hcatalog-unit/src/test/java/org/apache/hive/hcatalog/api/TestHCatClientNotification.java
@@ -83,7 +83,7 @@ public void createDatabase() throws Exception {
     assertEquals(firstEventId + 1, event.getEventId());
     assertTrue(event.getEventTime() >= startTime);
     assertEquals(HCatConstants.HCAT_CREATE_DATABASE_EVENT, event.getEventType());
-    assertNull(event.getDbName());
+    assertEquals("myhcatdb", event.getDbName());
     assertNull(event.getTableName());
     assertTrue(event.getMessage().matches("\\{\"eventType\":\"CREATE_DATABASE\",\"server\":\"\"," +
         "\"servicePrincipal\":\"\",\"db\":\"myhcatdb\",\"timestamp\":[0-9]+}"));
@@ -124,7 +124,7 @@ public void createTable() throws Exception {
     assertTrue(event.getEventTime() >= startTime);
     assertEquals(HCatConstants.HCAT_CREATE_TABLE_EVENT, event.getEventType());
     assertEquals(dbName, event.getDbName());
-    assertNull(event.getTableName());
+    assertEquals("hcatcreatetable", event.getTableName());
     assertTrue(event.getMessage().matches("\\{\"eventType\":\"CREATE_TABLE\",\"server\":\"\"," +
         "\"servicePrincipal\":\"\",\"db\":\"default\",\"table\":\"hcatcreatetable\",\"timestamp\":[0-9]+}"));
   }
diff --git a/itests/hcatalog-unit/src/test/java/org/apache/hive/hcatalog/listener/TestDbNotificationListener.java b/itests/hcatalog-unit/src/test/java/org/apache/hive/hcatalog/listener/TestDbNotificationListener.java
index c10fd7f..5bce2b1 100644
--- a/itests/hcatalog-unit/src/test/java/org/apache/hive/hcatalog/listener/TestDbNotificationListener.java
+++ b/itests/hcatalog-unit/src/test/java/org/apache/hive/hcatalog/listener/TestDbNotificationListener.java
@@ -89,7 +89,7 @@ public void createDatabase() throws Exception {
     assertEquals(firstEventId + 1, event.getEventId());
     assertTrue(event.getEventTime() >= startTime);
     assertEquals(HCatConstants.HCAT_CREATE_DATABASE_EVENT, event.getEventType());
-    assertNull(event.getDbName());
+    assertEquals("mydb", event.getDbName());
     assertNull(event.getTableName());
     assertTrue(event.getMessage().matches("\\{\"eventType\":\"CREATE_DATABASE\",\"server\":\"\"," +
         "\"servicePrincipal\":\"\",\"db\":\"mydb\",\"timestamp\":[0-9]+}"));
@@ -133,7 +133,7 @@ public void createTable() throws Exception {
     assertTrue(event.getEventTime() >= startTime);
     assertEquals(HCatConstants.HCAT_CREATE_TABLE_EVENT, event.getEventType());
     assertEquals("default", event.getDbName());
-    assertNull(event.getTableName());
+    assertEquals("mytable", event.getTableName());
     assertTrue(event.getMessage().matches("\\{\"eventType\":\"CREATE_TABLE\",\"server\":\"\"," +
         "\"servicePrincipal\":\"\",\"db\":\"default\",\"table\":\"mytable\",\"timestamp\":[0-9]+}"));
   }
-- 
1.7.9.5

